### PR TITLE
caster-soundboard: init at unstable-2019-9-28

### PR DESCRIPTION
**Pull request by IQubic**
###### Motivation for this change
caster-soundboard is a great application, and I noticed it wasn't in nixpkgs, so I packaged it up.

###### Things done
Added a derivation for caster-soundboard in pkg/audio
Added caster-soundboard to all-packages.nix
Added iqubic to the maintainers

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/caster-soundboard/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  };
+  sourceRoot = "source/CasterSoundboard";
+
+  buildInputs = [
+    qtmultimedia
+    gst_all_1.gst-plugins-base
+    gst_all_1.gst-plugins-good
+    gst_all_1.gst-plugins-bad
+    gst_all_1.gst-plugins-ugly
+  ];
```
Comment (worldofpeace) [inline pkgs/applications/audio/caster-soundboard/default.nix:21]: I believe you should also need `gst_all_1.gstreamer`?

File: pkgs/applications/audio/caster-soundboard/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  nativeBuildInputs = [ qmake ];
+
+  enableParallelBuilding = true;
+
+  qtWrapperArgs = [
+    ''--prefix GST_PLUGIN_SYSTEM_PATH_1_0 : "$GST_PLUGIN_SYSTEM_PATH_1_0"''
+  ];
+
+  meta = {
```
Comment (worldofpeace) [inline pkgs/applications/audio/caster-soundboard/default.nix:30]:
```suggestion
  meta = with stdenv.lib; {
```

File: pkgs/applications/audio/caster-soundboard/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    description = "A soundboard for hot-keying and playing back sounds.";
+    homepage = "https://github.com/JupiterBroadcasting/CasterSoundboard";
+    maintainers = [ stdenv.lib.maintainers.iqubic ];
```
Comment (worldofpeace) [inline pkgs/applications/audio/caster-soundboard/default.nix:33]:
```suggestion
    maintainers = [ maintainers.iqubic ];
```

File: pkgs/applications/audio/caster-soundboard/default.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    license = stdenv.lib.licenses.lgpl3;
```
Comment (worldofpeace) [inline pkgs/applications/audio/caster-soundboard/default.nix:34]:
```suggestion
    license = licenses.lgpl3;
```

File: pkgs/applications/audio/caster-soundboard/default.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    platforms = stdenv.lib.platforms.linux;
```
Comment (worldofpeace) [inline pkgs/applications/audio/caster-soundboard/default.nix:35]:
```suggestion
    platforms = platforms.linux;
```

File: pkgs/applications/audio/caster-soundboard/default.nix [hunk 6]
```diff
@@ -0,0 +1,37 @@
+{ stdenv, fetchFromGitHub, mkDerivation
+, qmake, qtmultimedia, gst_all_1 }:
+
+mkDerivation {
+  name = "caster-soundboard";
+  version = "unstable-2019-9-28";
+  src = fetchFromGitHub {
```
Comment (worldofpeace) [inline pkgs/applications/audio/caster-soundboard/default.nix:7]:
```suggestion

   src = fetchFromGitHub {
```

a newline

File: pkgs/applications/audio/caster-soundboard/default.nix [hunk 7]
```diff
@@ -0,0 +1,37 @@
+{ stdenv, fetchFromGitHub, mkDerivation
+, qmake, qtmultimedia, gst_all_1 }:
+
+mkDerivation {
+  name = "caster-soundboard";
```
Comment (worldofpeace) [inline pkgs/applications/audio/caster-soundboard/default.nix:5]:
```suggestion
  pname = "caster-soundboard";
```

File: pkgs/applications/audio/caster-soundboard/default.nix [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+  version = "unstable-2019-9-28";
+  src = fetchFromGitHub {
+    owner = "JupiterBroadcasting";
+    repo = "CasterSoundboard";
+    rev = "1a84315b3a3d4bdca1a4b784bd19460637f44438";
+    sha256 = "1npjcxw304hv8ihwv13cslbn2rcd6iqnknyqz94rdi496d1kfwv1";
+  };
+  sourceRoot = "source/CasterSoundboard";
```
Comment (worldofpeace) [inline pkgs/applications/audio/caster-soundboard/default.nix:13]:
```suggestion

  sourceRoot = "source/CasterSoundboard";
```

a newline

File: pkgs/applications/audio/caster-soundboard/default.nix [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+  nativeBuildInputs = [ qmake ];
+
+  enableParallelBuilding = true;
+
+  qtWrapperArgs = [
+    ''--prefix GST_PLUGIN_SYSTEM_PATH_1_0 : "$GST_PLUGIN_SYSTEM_PATH_1_0"''
+  ];
+
+  meta = {
+    description = "A soundboard for hot-keying and playing back sounds.";
```
Comment (worldofpeace) [inline pkgs/applications/audio/caster-soundboard/default.nix:31]:
```suggestion
    description = "A soundboard for hot-keying and playing back sounds";
```

no periods at the end of `description`

**Review by worldofpeace — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/caster-soundboard/default.nix (hunk 1 above)
Comment (worldofpeace) [reply pkgs/applications/audio/caster-soundboard/default.nix:21]: Also `qtbase`

**Review by IQubic — COMMENTED** [review]
This is my first time submitting any PRs to nixpkgs. I'm sorry for getting a lot of the little formatting details slightly wrong. I'll try to do better in the future. All in all, I think these changes are acceptable, and should be made to the commits I've already done.

**Comment by worldofpeace** [comment]
> This is my first time submitting any PRs to nixpkgs. I'm sorry for getting a lot of the little formatting details slightly wrong. I'll try to do better in the future. All in all, I think these changes are acceptable, and should be made to the commits I've already done.

Cool, that means you'll supply the changes in the coming day/s?

**Comment by IQubic** [comment]
@worldofpeace I wasn't aware that I had to supply the changes myself, I thought that anyone with push access to the repo could make the changes for me. Sorry about that. I'll get on this right away.

File: maintainers/maintainer-list.nix [hunk 10]
```diff
@@ -8869,4 +8869,10 @@
     github = "cpcloud";
     githubId = 417981;
   };
+  iqubic = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:8872]: This must be sorted alphabetical

File: pkgs/applications/audio/caster-soundboard/default.nix [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+    qtmultimedia
+    gst_all_1.gst-plugins-base
+    gst_all_1.gst-plugins-good
+    gst_all_1.gst-plugins-bad
+    gst_all_1.gst-plugins-ugly
+  ];
+  nativeBuildInputs = [ qmake ];
+
+  enableParallelBuilding = true;
+
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/caster-soundboard/default.nix:27]:
```suggestion
```
done by qmake

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by davidak** [comment]
The project looks abandoned. No commit since [Sep 28, 2019](https://github.com/JupiterBroadcasting/CasterSoundboard/commit/1a84315b3a3d4bdca1a4b784bd19460637f44438).

**Comment by Janik-Haag** [comment]
Will close the pr since the project is almost definitively stale.

**Pull request closed without merging**